src/walk.ts

```diff
@@ -317,6 +317,9 @@ export function walkDocument<T>(opts: {
       ruleId: string,
       severity: ProblemSeverity,
     ) {
+      if (!node) {
```

Comment:
check for `null` or `undefined` specifically, we may want to visit `0` or the empty string `""`.

Also, please, add some comment explaining this return (this is not obvious that type mismatch errors will be spotted by `spec` rule on the parent object.

Also, please check for `'scalar'` type here and visit scalar types regardless of value:

https://github.com/Redocly/openapi-cli/blob/master/src/walk.ts#L242